### PR TITLE
[Containerapp] `az containerapp create`: Fix BadRequest Error about clientType with --bind

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp create --bind

There is a bug with command:
az containerapp create -n bind-test-xinyu -g  arc-appint-forxinyu12-rg0 --environment $envId --bind postgres --debug

It contains `"clientType": null`, which cause following BadRequest error:
```
"serviceBinds": [{"serviceId": "/subscriptions/xxxxx/resourceGroups/arc-appint-forxinyu12-rg0/providers/Microsoft.App/containerapps/postgres", "name": "postgres", "clientType": null}]}
```
```
Bad Request({"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"One or more validation errors occurred.","status":400,"traceId":"00-695d93a999533b4b165734543bdaa919-1e3b87573c3523ee-01","errors":{"$[0].clientType":["The JSON value could not be converted to Microsoft.ContainerApps.ApiResources.Models.ServiceClientType. Path: $[0].clientType | LineNumber: 0 | BytePositionInLine: 196."]}})
az_command_data_logger: Bad Request({"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"One or more validation errors occurred.","status":400,"traceId":"00-695d93a999533b4b165734543bdaa919-1e3b87573c3523ee-01","errors":{"$[0].clientType":["The JSON value could not be converted to Microsoft.ContainerApps.ApiResources.Models.ServiceClientType. Path: $[0].clientType | LineNumber: 0 | BytePositionInLine: 196."]}})
```
The fix this not set `"clientType"` in the requestBody when clientType is not in the arg_dict.  

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

**Comment by Greedygre** [comment]
Hi @wangzelin007 

The CI failed with error:
Looks not relate to our change and extension, could you help to handle it?

```
ERROR: test_ref_doc_command-change (__main__.IndexRefDocs.test_ref_doc_command-change)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/connectionpool.py", line 467, in _make_request
    self._validate_conn(conn)
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/connectionpool.py", line 1096, in _validate_conn
    conn.connect()
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/connection.py", line 642, in connect
    sock_and_verified = _ssl_wrap_socket_and_match_hostname(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/connection.py", line 782, in _ssl_wrap_socket_and_match_hostname
    ssl_sock = ssl_wrap_socket(
               ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 470, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 514, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/ssl.py", line 517, in wrap_socket
    return self.sslsocket_class._create(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/ssl.py", line 1108, in _create
    self.do_handshake()
  File "/opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/ssl.py", line 1379, in do_handshake
    self._sslobj.do_handshake()
TimeoutError: [Errno 110] Connection timed out

```

**Review by bgashirabake — APPROVED** [review]
LGTM 🚀

File: src/containerapp/azext_containerapp/_utils.py [hunk 1]
```diff
@@ -81,8 +81,9 @@ def process_service(cmd, resource_list, service_name, arg_dict, subscription_id,
                 service_bind = {
                     "serviceId": containerapp_def["id"],
                     "name": binding_name,
-                    "clientType": arg_dict.get("clientType")
                 }
+                if arg_dict.get("clientType"):
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_utils.py:85]: then how can customer clear clilent type?

Comment (Greedygre) [reply src/containerapp/azext_containerapp/_utils.py:85]: If the Cx want to clean the clientType, input `clientType=none`

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py [hunk 2]
```diff
@@ -704,6 +704,51 @@ def test_containerapp_up_dapr_e2e(self, resource_group):
...
+                ca_name, resource_group, env_id, redis_ca_name, env_rg, postgres_ca_name, env_rg), expect_failure=False, checks=[
+                JMESPathCheck('properties.provisioningState', "Succeeded"),
+                JMESPathCheck('length(properties.template.serviceBinds)', 2),
+                JMESPathCheck('properties.template.serviceBinds[0].name', redis_ca_name),
+                JMESPathCheck('properties.template.serviceBinds[0].clientType', "dotnet"),
+                JMESPathCheck('properties.template.serviceBinds[1].name', postgres_ca_name),
+                JMESPathCheck('properties.template.serviceBinds[1].clientType', "none"),
+            ])
+
+        self.cmd(
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:742]: please also add test for update related scenario

Comment (Greedygre) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:754]: Added.

**Review by Juliehzl — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 

Could you help to review and merge this PR? Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@Greedygre Please resolve this conflict

**Pull request merged**